运行记录：

	hadoop@Master:~/xubo/ref/GRCH38L1Index/pe$ bwa mem ../GRCH38chr1L3556522.fasta GRCH38chr1L3556522N1000000L100paired1.fastq GRCH38chr1L3556522N1000000L100paired2.fastq >GRCH38chr1L3556522N1000000L100paired12.sam
	[M::bwa_idx_load_from_disk] read 0 ALT contigs
	[M::process] read 100000 sequences (10000000 bp)...
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44602, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.79, 49.84)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.011 CPU sec, 37.988 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44446, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 498, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (331, 666)
	[M::mem_pestat] mean and std.dev: (498.57, 49.70)
	[M::mem_pestat] low and high boundaries for proper pairs: (264, 733)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.287 CPU sec, 38.187 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44553, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (499.09, 49.89)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.447 CPU sec, 37.335 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44490, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.78, 49.86)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.639 CPU sec, 37.529 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44537, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (499.18, 49.73)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.181 CPU sec, 37.063 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44493, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (466, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (332, 667)
	[M::mem_pestat] mean and std.dev: (499.19, 49.67)
	[M::mem_pestat] low and high boundaries for proper pairs: (265, 734)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.731 CPU sec, 37.645 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44395, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (499.19, 49.91)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.868 CPU sec, 38.783 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44428, 1, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.75, 49.92)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.643 CPU sec, 39.199 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44455, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.90, 49.75)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.059 CPU sec, 38.147 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44497, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.75, 49.95)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.032 CPU sec, 38.342 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44452, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (466, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (332, 667)
	[M::mem_pestat] mean and std.dev: (499.25, 49.76)
	[M::mem_pestat] low and high boundaries for proper pairs: (265, 734)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.383 CPU sec, 41.257 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44624, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.90, 49.83)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.706 CPU sec, 37.771 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44548, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 498, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (331, 666)
	[M::mem_pestat] mean and std.dev: (498.50, 49.73)
	[M::mem_pestat] low and high boundaries for proper pairs: (264, 733)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.009 CPU sec, 38.306 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44369, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (466, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (332, 667)
	[M::mem_pestat] mean and std.dev: (499.29, 49.65)
	[M::mem_pestat] low and high boundaries for proper pairs: (265, 734)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.027 CPU sec, 38.429 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44525, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (331, 666)
	[M::mem_pestat] mean and std.dev: (498.71, 49.67)
	[M::mem_pestat] low and high boundaries for proper pairs: (264, 733)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.977 CPU sec, 39.461 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44586, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (331, 666)
	[M::mem_pestat] mean and std.dev: (498.80, 49.59)
	[M::mem_pestat] low and high boundaries for proper pairs: (264, 733)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.294 CPU sec, 40.004 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44519, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (464, 498, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (328, 668)
	[M::mem_pestat] mean and std.dev: (498.21, 50.11)
	[M::mem_pestat] low and high boundaries for proper pairs: (260, 736)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.618 CPU sec, 43.286 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44469, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (329, 669)
	[M::mem_pestat] mean and std.dev: (498.90, 49.89)
	[M::mem_pestat] low and high boundaries for proper pairs: (261, 737)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.862 CPU sec, 39.607 real sec
	[M::process] read 100000 sequences (10000000 bp)...
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44503, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (466, 499, 533)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (332, 667)
	[M::mem_pestat] mean and std.dev: (498.94, 49.63)
	[M::mem_pestat] low and high boundaries for proper pairs: (265, 734)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 38.381 CPU sec, 41.691 real sec
	[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44510, 0, 0)
	[M::mem_pestat] skip orientation FF as there are not enough pairs
	[M::mem_pestat] analyzing insert size distribution for orientation FR...
	[M::mem_pestat] (25, 50, 75) percentile: (465, 499, 532)
	[M::mem_pestat] low and high boundaries for computing mean and std.dev: (331, 666)
	[M::mem_pestat] mean and std.dev: (498.66, 49.88)
	[M::mem_pestat] low and high boundaries for proper pairs: (264, 733)
	[M::mem_pestat] skip orientation RF as there are not enough pairs
	[M::mem_pestat] skip orientation RR as there are not enough pairs
	[M::mem_process_seqs] Processed 100000 reads in 37.654 CPU sec, 41.754 real sec
	[main] Version: 0.7.13-r1126
	[main] CMD: bwa mem ../GRCH38chr1L3556522.fasta GRCH38chr1L3556522N1000000L100paired1.fastq GRCH38chr1L3556522N1000000L100paired2.fastq
	[main] Real time: 790.344 sec; CPU: 760.405 sec
	
	
flagstat：
	
	hadoop@Master:~/xubo/ref/GRCH38L1Index/pe$ samtools flagstat GRCH38chr1L3556522N1000000L100paired12.sam 
	2000000 + 0 in total (QC-passed reads + QC-failed reads)
	0 + 0 secondary
	0 + 0 supplementary
	0 + 0 duplicates
	2000000 + 0 mapped (100.00% : N/A)
	2000000 + 0 paired in sequencing
	1000000 + 0 read1
	1000000 + 0 read2
	1999960 + 0 properly paired (100.00% : N/A)
	2000000 + 0 with itself and mate mapped
	0 + 0 singletons (0.00% : N/A)
	0 + 0 with mate mapped to a different chr
	0 + 0 with mate mapped to a different chr (mapQ>=5)
